### PR TITLE
1441 - Add migration guide

### LINKED ISSUE #1441: Docs: Migration Guide and Training
**Describe the bug**

Getting some requests for the title items so thinking of some ideas to collect this.

- [ ] Work with a team and record steps for migration
- [ ] Improve notes on implementing component to component
- [ ] Mention side by side migration and general quirks and things to know

Training (Think new developer joining Infor

- [ ] What is Ids?
- [ ] Resources / Links and locations
- [ ] What to think about...

### PR DESCRIPTION
**Pull request by tmcconechy**
**Explain the details for making this change. What existing problem does the pull request solve?**

Adds more root docs, a doc index page, contribution guide and a migration guide. The main stratgey in the guide is to map out intelligent sections with useful info and link to it. Rather than repeat info in various places

**Related github/jira issue (required)**:
Fixes #1441

**Steps necessary to review your pull request (required)**:
- make any comments or additons at any time and make PR comments
- browse the readme https://github.com/infor-design/enterprise-wc/tree/1441-migration-guide
- browse the migration  https://github.com/infor-design/enterprise-wc/blob/1441-migration-guide/doc/MIGRATION-GUIDE.md
- browse the contribution guide https://github.com/infor-design/enterprise-wc/blob/1441-migration-guide/doc/CONTRIBUTING.md

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

**Review by jdilag90 — APPROVED** [review]

**Comment by tmcconechy** [comment]
@EdwardCoyle open for PR if you have any additions either comment or additions either comment make additions on the branch. Thanks!

File: doc/MIGRATION-GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#  Migration Guide
+
+While everyone's path and and needs may vary, this guide is meant to provide a list of common questions and answers and notes that may be helpful while migrating from previous versions of IDS or using it for the first. This is meant to be a live guide. If you have suggestions please [make an issue](https://github.com/infor-design/enterprise-wc/issues) and even better [a pull request](./CONTRIBUTING.md).
+
+## What is a Web Component
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:5]:
```suggestion
## What is a Web Component?
```

File: doc/MIGRATION-GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+See the [browser support](../README.md#browser-support) section for this information.
+
+## Dependencies
+
+The ids-enterprise-wc web components no longer have any (dev) dependencies.
+
+- We dropped jQuery as a dependency, its no longer needed unless running old components side by side
+- We dropped d3 as a dependency, its no longer needed unless running old components side by side.
+
+## Installing into your framework
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:33]:
```suggestion
## Installation
```

File: doc/MIGRATION-GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+```js
+const myDatePicker = document.querySelector<IdsDatePicker>('ids-date-picker');
+myDatePicker.minuteInterval = 15;
+```
+
+## Where is my feature XYX
+
+All the features should be in the docs for each component. See the index for a full listing of [docs for each component](./DOCUMENTATION.md). If a setting is missing in the docs you can also consult the [source for the components](https://github.com/infor-design/enterprise-wc/tree/main/src/components). And [let us know](https://github.com/infor-design/enterprise-wc/issues/new/choose) or give us a pull request.
+
+Note that we did not add every feature from the previous version. We tried to select the most common / used features. We also wanted to deprecate some things that were confusing / wrong or badly done. If we missed one that you find important and need [let us know](https://github.com/infor-design/enterprise-wc/issues/new/choose) or give us a pull request.
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:91]:
```suggestion
Note that we did not add every feature from the previous version. We selected the most common / used features. We also wanted to deprecate some things that were confusing, incorrect, or did not provide an adequate user experience. If we missed a feature that you find important, let us know or [contribute it with a pull request](https://github.com/infor-design/enterprise-wc/issues/new/choose).
```

File: doc/MIGRATION-GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+- `properties` Refer to [js properties](https://developer.mozilla.org/en-US/docs/Glossary/Property/JavaScript). So all settings can also be referred to as properties. When used in Js they are done with camel case as is the js naming convention.  Note that any types can be used here and Typescript types should show in the intellisense if installed property. For example:
+
+```js
+const myDatePicker = document.querySelector<IdsDatePicker>('ids-date-picker');
+myDatePicker.minuteInterval = 15;
+```
+
+## Where is my feature XYX
+
+All the features should be in the docs for each component. See the index for a full listing of [docs for each component](./DOCUMENTATION.md). If a setting is missing in the docs you can also consult the [source for the components](https://github.com/infor-design/enterprise-wc/tree/main/src/components). And [let us know](https://github.com/infor-design/enterprise-wc/issues/new/choose) or give us a pull request.
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:89]:
```suggestion
All existing component features are documented in their respective `README.md` file. See the index for a full listing of [docs for each component](./DOCUMENTATION.md). If a setting is missing from the documentation, you can also consult the [source for the components](https://github.com/infor-design/enterprise-wc/tree/main/src/components). [Pull requests](https://github.com/infor-design/enterprise-wc/issues/new/choose) for missing documentation are welcome.
```

File: doc/MIGRATION-GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+## Example Projects
+
+We dont recommend any particular framework but we try to test with the most common ones and provide guidance on how to use the components. If we missed anything or you want to see a pattern example in a framework [make an issue](https://github.com/infor-design/enterprise-wc/issues/new/choose). As of now we have examples for the following frameworks click the link for guidance on each one. The example project can be found on [this repo](https://github.com/infor-design/enterprise-wc-examples), each example is in various levels of maturity.
+
+- **[Angular](https://github.com/infor-design/enterprise-wc-examples/tree/main/angular-ids-wc)**
+- **[React](https://github.com/infor-design/enterprise-wc-examples/tree/main/react-ids-wc)**
+- **[React Typescript](https://github.com/infor-design/enterprise-wc-examples/tree/main/react-ts-ids-wc)**
+- **[Vue JS](https://github.com/infor-design/enterprise-wc-examples/tree/main/vue-ids-wc)**
+- **[Svelte Kit](https://github.com/infor-design/enterprise-wc-examples/tree/main/sveltekit-ids-wc)**
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:116]:
```suggestion
- **[SvelteKit](https://github.com/infor-design/enterprise-wc-examples/tree/main/sveltekit-ids-wc)**
```

File: doc/MIGRATION-GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+Once you import the script the custom element will self-register and then you can just use like the markup shown in the [docs for each component](./DOCUMENTATION.md).
+
+## Documentation
+
+Each component has an `md` file for documentation. See the index for a full listing of [docs for each component](./DOCUMENTATION.md)
+
+## Side By Side Migration Approach
+
+Side by Side migration to us means you can the older version of components with this version of the components. Then swap out components side by side as both can run together in the same page. Note: that not all possibilities have been tested, but if you run into an issue we can possibly fix it, just raise an issue.
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:66]:
```suggestion
The Web Component library has been designed to run side by side together with the jQuery/Angular components on the same page. "Side by Side Migration", as we define it, means starting with the jQuery/Angular components while slowly swapping them out for the Web Components over time.  Please note that not all combinations have been thoroughly tested.  If you run into any problems, please [raise an issue on Github](https://github.com/infor-design/enterprise-wc/issues/new/choose).
```

File: doc/MIGRATION-GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+## Installing into your framework
+
+The components are available via npm/yarn:
+
+```bash
+npm install --save ids-enterprise-wc@latest
+```
+
+After that the files can be seen in `node_modules/ids-enterprise-wc`. Then just import the components as needed. You can either import all component in one import via the `enterprise-wc.js` file. A better method is to import each and only each component you actually use, if doing a side by side approach this is especially important.
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:41]:
```suggestion
After installing, the files can be seen in `node_modules/ids-enterprise-wc`. Then, simply import the components as needed. You can either import all component in one import via the `enterprise-wc.js` file, or (even better) import each component individually.  Importing individually is better for side by side migration, and is better supported by tree-shaking operations in popular build tools like Webpack.
```

File: doc/MIGRATION-GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+Go Component By Component: Most components have a side-by-side example to test it you can review on the demo site https://main.wc.design.infor.com/ then click each component and see if there is a side by side page that show cases the two components can work side by side. But also you might want to work component by component and replace all uses of it, for example all old IDS buttons to `<ids-button>`.
+
+## What are settings / attributes / properties
+
+To clarify we use these terms in the following way:
+
+- `settings` In general everything is a setting as a generic term. For example `minute interval` is a setting on date picker for the time minutes picker. Almost every component has an array of settings mentioned in the docs. In case we missed one in the docs note that they can always been seen in each component source in a `attributes` section like [this component](https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-alert/ids-alert.ts#L49).
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:78]:
```suggestion
- `settings` In general, attributes and properties are all technically "settings".  Almost every component has an array of settings mentioned in the docs. If documentation on a setting is missing, each source file has an `attributes` section like [this component](https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-alert/ids-alert.ts#L49) that lists each one.  Additionally, any mixins passed into a component can have their own attributes.  The mixins that help construct a component [can be found near the top](https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-module-nav/ids-module-nav-item.ts#L10-L16) of each source file, below the imports.
```

File: doc/MIGRATION-GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+- `attributes` Refer to either the html attribute name. Or the fact that the [web component spec](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements#responding_to_attribute_changes) calls the attributes. So all settings can also be referred to as attributes. When used in markup they are done with kebab case as is the html spec. For example `<ids-date-picker minute-interval="15"`. Note that primitive types can have html attribute style settings, any object settings must be done with Js.
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:79]:
```suggestion
- `attributes` Refers to the HTML attribute that targets a particular setting.  The [Web Component specification](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements#responding_to_attribute_changes) calls these "attributes". Most IDS component settings are configurable using a kebab-case HTML attribute if they correspond to a primitive type like a string, number or boolean (For example `<ids-date-picker minute-interval="15"`). Note that only primitive types can be passed as an HTML attribute, and more complex types like objects or arrays must be defined as JS properties.
```

File: doc/MIGRATION-GUIDE.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+- `properties` Refer to [js properties](https://developer.mozilla.org/en-US/docs/Glossary/Property/JavaScript). So all settings can also be referred to as properties. When used in Js they are done with camel case as is the js naming convention.  Note that any types can be used here and Typescript types should show in the intellisense if installed property. For example:
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:80]:
```suggestion
- `properties` Refers to [JavaScript properties](https://developer.mozilla.org/en-US/docs/Glossary/Property/JavaScript) that target a particular setting. All IDS component settings are configurable using a "camelCase" property representing all JavaScript or TypeScript type.  If Typescript is used, types should show in your editor's intellisense if installed properly. For example:
```

File: doc/MIGRATION-GUIDE.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+Start Slow: Consider areas of the application that need improvement, or new areas or maybe an admin page that needs rework and has lower impact.
+
+Get New Features: Try to migrate to components that have new features that wont be added to the old component library. Like much cleaner way to use icons with `<ids-icon>` and a future icon font, or components like a new `scroll-view`, `data grid` virtual scroll, `stats`, new `ids-charts`, `ids-action-sheet`, new `ids-loading-indicator` styles and a much cleaner and better stand alone `ids-pager` to name a few.
+
+Go Component By Component: Most components have a side-by-side example to test it you can review on the demo site https://main.wc.design.infor.com/ then click each component and see if there is a side by side page that show cases the two components can work side by side. But also you might want to work component by component and replace all uses of it, for example all old IDS buttons to `<ids-button>`.
+
+## What are settings / attributes / properties
+
+To clarify we use these terms in the following way:
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:76]:
```suggestion
Not all properties will correspond to attributes, but all attributes are represented by properties.  To clarify, IDS uses these terms as described below:
```

File: doc/MIGRATION-GUIDE.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+Note that we did not add every feature from the previous version. We tried to select the most common / used features. We also wanted to deprecate some things that were confusing / wrong or badly done. If we missed one that you find important and need [let us know](https://github.com/infor-design/enterprise-wc/issues/new/choose) or give us a pull request.
+
+## Customizing and Themes
+
+We added a seperate section on ways you can [customize a component](./DOCUMENTATION.md). These include using themes, css parts, extending ect.
+
+## Component specific changes
+
+When migrating a component, review the documentation for the component your migrating. Note that each component doc contains a section called `Converting from Previous Versions (Breaking Changes)` which lists specific notes for each component. Again if we missed anything [make an issue](https://github.com/infor-design/enterprise-wc/issues/new/choose).
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:99]:
```suggestion
When migrating a component, closely review that component's documentation. Note that each component doc contains a section called `Converting from Previous Versions (Breaking Changes)` which lists specific notes for upgrading. If we missed anything, [please raise a Github issue](https://github.com/infor-design/enterprise-wc/issues/new/choose).
```

File: doc/MIGRATION-GUIDE.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+## Where is my feature XYX
+
+All the features should be in the docs for each component. See the index for a full listing of [docs for each component](./DOCUMENTATION.md). If a setting is missing in the docs you can also consult the [source for the components](https://github.com/infor-design/enterprise-wc/tree/main/src/components). And [let us know](https://github.com/infor-design/enterprise-wc/issues/new/choose) or give us a pull request.
+
+Note that we did not add every feature from the previous version. We tried to select the most common / used features. We also wanted to deprecate some things that were confusing / wrong or badly done. If we missed one that you find important and need [let us know](https://github.com/infor-design/enterprise-wc/issues/new/choose) or give us a pull request.
+
+## Customizing and Themes
+
+We added a seperate section on ways you can [customize a component](./DOCUMENTATION.md). These include using themes, css parts, extending ect.
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:95]:
```suggestion
We added a separate section on ways you can [customize a component](./DOCUMENTATION.md). These include using themes, css parts, extending etc.
```

File: doc/MIGRATION-GUIDE.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+## Getting Help
+
+We :heart: working with developers and we have a few channels for questions or support:
+
+- [Contributing](./CONTRIBUTING.md) with bug reports and pull requests and suggestions
+- For asking general questions or searching for other questions use or [Ms Teams Channel](https://teams.microsoft.com/l/team/19%3A2b0c9ce520b0481a9ce115f0ca4a326f%40thread.skype/conversations?groupId=4f50ef7d-e88d-4ccb-98ca-65f26e57fe35&tenantId=457d5685-0467-4d05-b23b-8f817adda47c)
+
+## Example Projects
+
+We dont recommend any particular framework but we try to test with the most common ones and provide guidance on how to use the components. If we missed anything or you want to see a pattern example in a framework [make an issue](https://github.com/infor-design/enterprise-wc/issues/new/choose). As of now we have examples for the following frameworks click the link for guidance on each one. The example project can be found on [this repo](https://github.com/infor-design/enterprise-wc-examples), each example is in various levels of maturity.
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:110]:
```suggestion
We don't recommend any particular framework, but we try to test with the most common ones and provide guidance on how to integrate. If we missed anything, or you want to see a pattern example in a framework, [please raise a Github issue](https://github.com/infor-design/enterprise-wc/issues/new/choose).  See the links below for guidance and integration examples of the frameworks we currently support. The example project can be found on [this repo](https://github.com/infor-design/enterprise-wc-examples), each example is in various levels of maturity.
```

File: doc/MIGRATION-GUIDE.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
+#  Migration Guide
+
+While everyone's path and and needs may vary, this guide is meant to provide a list of common questions and answers and notes that may be helpful while migrating from previous versions of IDS or using it for the first. This is meant to be a live guide. If you have suggestions please [make an issue](https://github.com/infor-design/enterprise-wc/issues) and even better [a pull request](./CONTRIBUTING.md).
```
Comment (EdwardCoyle) [inline doc/MIGRATION-GUIDE.md:3]:
```suggestion
While everyone's path and and needs may vary, this guide is meant to provide a list of common questions and answers and notes that may be helpful while migrating from previous versions of IDS, or using it for the first time. This is meant to be a live guide. If you have suggestions please [raise an issue on Github](https://github.com/infor-design/enterprise-wc/issues) or even better [a pull request with your contribution](./CONTRIBUTING.md).
```

**Review by EdwardCoyle — COMMENTED** [review]
Various grammar and sentence structure suggestions incoming

**Pull request merged**